search/collector/topn.go

```diff
@@ -418,6 +426,68 @@ func (hc *TopNCollector) Total() uint64 {
 	return hc.total
 }
 
+func (hc *TopNCollector) SetKNNCollectorStore(kArray []int64, scoreCorrector func(docMatch *search.DocumentMatch)) {
+	// background - assuming D segments exists in the index (D is a highly variable number)
+	// the vector reader returns the top K results for each segment
+	// resulting in D * K hits matching the KNN query. With M KNN queries,
+	// the total number of results matching the KNN queries is at max D * M * K
+	// the goal is to ensure that the behavior remains the same as observed
+	// in a single segment scenario - M * K results are returned for M KNN queries
+	//
+	// the scenario becomes complex when pagination is involved - since the pagination is done on the total score
+	// the total score is a combination of TF-IDF score and KNN scores and as the document matches
+	// are streamed from the compound searcher across Next() calls,
+	// attempting to retain the top K elements for each KNN query with accurate pagination
+	// becomes a challenge - since for each KNN query we need to reduce the KNN score
+	// of D*K - K = K*(D-1) hits to 0, which reduces the total score of the document
+	// which inturn messes up the pagination - since the total score is used for pagination
+	//
+	// this problem is a constrained equivalent to the problem of maintaining top N elements from an infinite
+	// stream of objects where the stream can contain duplicates, (wherein we need to
+	// update the value of the duplicate object received in the stream), with the additional
+	// constraint that the space complexity of the solution should be O(N) and not infinite
+	//
+	// idea:-
+	// keep a min-heap for each KNN query 	- this is because we need to maintain top K elements for each KNN query
```

Comment:
i'm a little confused, it's a max heap you're constructing right? because i see below that 
```
if i.Score < j.Score {
	return 1
}
return -1
```
 
and the heap comparator logic indicates it returns 1 if i.Score < j.Score (which is opposite) of the logic in sort.go